slowquant/qiskit_interface/interface.py

```diff
@@ -217,17 +217,31 @@ def _sampler_quantum_expectation_value(self, op: FermionicOperator, run_paramete
         values = 0.0
         observables = self.op_to_qbit(op)
 
+        # Loop over all clique Paulies
+        cliques = make_cliques(observables.paulis)
+        distributions = {}
+        for clique_pauli in cliques.keys():
+            dist = self._sampler_distributions(Pauli(clique_pauli), run_parameters)
+            # It is wasteful to store the distribution per Pauli instead of per Clique,
```

Comment:
Do you mean you store per Pauli string that is encompassed in a clique? You dont mean you store the actual single Pauli gate?

Reply:
I mean that I store the distribution for a clique per Pauli string inside the clique.

If I have the clique {XX: [XI, IX, XX]}, then instead of storing the bitstring distribution only for the clique index "XX", it is stored for all the Pauli strings XI, IX, XX.

This is ofc wasteful in storage, but it makes the loop in line 233 more simple